src/main/java/com/_up/megastore/controllers/interfaces/IProductController.java

```diff
@@ -30,4 +30,8 @@ ProductResponse updateProduct(
             @RequestPart @Nullable MultipartFile multipartFile
     );
 
+    @PatchMapping("/{productId}/restore")
+    @ResponseStatus(HttpStatus.OK)
+    ProductResponse restoreProduct(@PathVariable UUID productId);
```

Comment:
Si bien es cierto que para darlos de alta solo tenemos que modificar un atributo, no está del todo bien que el método HTTP sea un Patch.

Cuando nosotros definimos un método HTTP, queremos dar a entender la razón del endpoint, en este caso no queremos indicar "actualizamos un atributo" sino "damos de alta una entidad que estaba eliminada", lo cual es una acción. Los endpoints de una acción se definen por método POST por lo que lo cambiaria de la siguiente manera

```suggestion
    @PostMapping("/{productId}/restore")
    @ResponseStatus(HttpStatus.OK)
    ProductResponse restoreProduct(@PathVariable UUID productId);
```

Esto es así como definimos la eliminación de un recurso mediante un borrado lógico, no usamos PATCH sino que DELETE, porque la intención es eliminar no actualizar un atributo